### DISCUSSION TITLE
ImportError: cannot import name 'Config' from 'mmcv' (unknown location) in ./tools/misc/print_config.py

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmdetection#8984] [Q&A]
mmdetection3.0版本安装后，出现这个怎么弄？
How do I get this after installing mmdetection version 3.0?
```shell
Traceback (most recent call last):
  File "./tools/misc/print_config.py", line 5, in <module>
    from mmcv import Config, DictAction
ImportError: cannot import name 'Config' from 'mmcv' (unknown location)
```

**Comment by zhangzscn** [comment]
Package            Version   Editable project location
------------------ --------- -----------------------------
addict             2.4.0
brotlipy           0.7.0
certifi            2022.6.15
cffi               1.15.1
charset-normalizer 2.1.0
click              8.1.3
colorama           0.4.5
coloredlogs        15.0.1
commonmark         0.9.1
contourpy          1.0.5
cryptography       37.0.2
cycler             0.11.0
flatbuffers        2.0
fonttools          4.37.4
humanfriendly      10.0
idna               3.3
importlib-metadata 5.0.0
kiwisolver         1.4.4
Markdown           3.4.1
matplotlib         3.6.0
mkl-fft            1.3.1
mkl-random         1.2.2
mkl-service        2.4.0
mmcv               2.0.0rc1
mmdet              3.0.0rc1  /data/private/zzs/mmdetection
mmengine           0.1.0
model-index        0.1.11
mpmath             1.2.1
numpy              1.23.1
onnxruntime        1.12.1
opencv-python      4.6.0.66
openmim            0.3.2
ordered-set        4.1.0
packaging          21.3
pandas             1.5.0
Pillow             9.2.0
pip                22.2.2
pycocotools        2.0.5
pycparser          2.21
Pygments           2.13.0
pyOpenSSL          22.0.0
pyparsing          3.0.9
PySocks            1.7.1
python-dateutil    2.8.2
pytz               2022.4
PyYAML             6.0
requests           2.28.1
rich               12.6.0
setuptools         65.4.1
six                1.16.0
sympy              1.10.1
tabulate           0.9.0
termcolor          2.0.1
terminaltables     3.1.10
torch              1.12.1
torchaudio         0.12.1
torchvision        0.13.1
typing_extensions  4.3.0
urllib3            1.26.11
wheel              0.37.1
yapf               0.32.0
zipp               3.8.1

**Comment by zhangzscn** [comment]
conda create --name openmmlab python=3.8 -y
conda activate openmmlab
conda install pytorch torchvision torchaudio cudatoolkit=11.6 -c pytorch -c conda-forge
pip install -U openmim
mim install mmengine
mim install "mmcv>=2.0.0rc1"
git clone https://github.com/open-mmlab/mmdetection.git -b dev-3.x
cd mmdetection
pip install -v -e .

**Reply by vansin** [reply]
please provide the python  script you run such as 

```shell
python tools/train.py xxxx.py
```

您的 tools 的脚本是不是用的 mmdetection 2.x 版本的 ？
Is your tools script using mmdetection version 2.x?

**Reply by zhangzscn** [reply]
python ./tools/misc/print_config.py configs/yolact/yolact_r50_8xb8-55e_coco.py

**Reply by zhangzscn** [reply]
./tools/dist_train.sh configs/yolact/yolact_r50_8xb8-55e_coco.py 8


/opt/conda/envs/zzs_openmmlab3/lib/python3.8/site-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  warnings.warn(
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
Traceback (most recent call last):
  File "./tools/train.py", line 12, in <module>
    from mmdet.utils import register_all_modules
  File "/data/private/zzs/mmdetection/mmdet/__init__.py", line 10, in <module>
    mmcv_version = digit_version(mmcv.__version__)
AttributeError: module 'mmcv' has no attribute '__version__'
Traceback (most recent call last):

**Reply by zhangzscn** [reply]
分支已经切换到dev-3.x

**Reply by vansin** [reply]
谢谢您公测的贡献，这个脚本确实有问题，没有更新。
Thank you for your contribution to the public test, this script does have problems and is not updated.

您可以将 mmcv 替换成 mmengine
You can replace mmcv with mmengine

welcome pr

```python 
from mmengine import Config, DictAction
```

**Reply by zhangzscn** [reply]
./tools/dist_train.sh 这个脚本也有问题

**Reply by vansin** [reply]
> ./tools/dist_train.sh 这个脚本也有问题

详细说说？
Tell us more about it?

**Reply by zhangzscn** [reply]
运行 ./tools/dist_train.sh configs/yolact/yolact_r50_8xb8-55e_coco.py 8 命令
出现下面的错误
/opt/conda/envs/zzs_openmmlab3/lib/python3.8/site-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  warnings.warn(
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
Traceback (most recent call last):
  File "./tools/train.py", line 12, in <module>
    from mmdet.utils import register_all_modules
  File "/data/private/zzs/mmdetection/mmdet/__init__.py", line 10, in <module>
    mmcv_version = digit_version(mmcv.__version__)
AttributeError: module 'mmcv' has no attribute '__version__'
Traceback (most recent call last):
  File "./tools/train.py", line 12, in <module>
    from mmdet.utils import register_all_modules
  File "/data/private/zzs/mmdetection/mmdet/__init__.py", line 10, in <module>
    mmcv_version = digit_version(mmcv.__version__)
AttributeError: module 'mmcv' has no attribute '__version__'
Traceback (most recent call last):
